src/coreclr/src/jit/optcse.cpp

```diff
@@ -1578,12 +1578,12 @@ class CSE_Heuristic
         sortTab          = nullptr;
         sortSiz          = 0;
 
-#ifdef _TARGET_XARCH_
+#ifdef _TARGET_X86_
         if (m_pCompiler->compLongUsed)
```

Comment:
I'm not sure why this is needed at all, much less x86 only. Can you explain? It might make sense to count 2 regs for each long lclVar but I'm not sure why this just adds one.

Reply:
It was needed by the old x86 tree based codegen.
It may no longer be needed with the RyuJIT based x86 codegen.

I will try removing it.

Reply:
Removing this has positive results for x86